### PR TITLE
Backport Things to 1.20.1

### PR DESCRIPTION
**Pull request by HowardZHY**
Not tested yet

### CODE REVIEW

**Comment by HowardZHY** [comment]
Now tested, I think you can use Beta channel on Modrinth

File: src/main/java/de/florianmichael/viafabricplus/definition/ClientsideFixes.java [hunk 1]
```diff
@@ -60,7 +62,6 @@ public class ClientsideFixes {
      * Contains the armor points of all armor items in legacy versions (<= 1.8.x)
      */
     private final static Map<Item, Integer> LEGACY_ARMOR_POINTS = new HashMap<>();
-
```
Comment (FlorianMichael) [inline src/main/java/de/florianmichael/viafabricplus/definition/ClientsideFixes.java:63]: Add the newline back

File: src/main/java/de/florianmichael/viafabricplus/definition/authlib/KeyPairResponse1_19_0.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package de.florianmichael.viafabricplus.definition.authlib;
+import com.mojang.authlib.yggdrasil.response.KeyPairResponse;
+import java.nio.ByteBuffer;
+/*
+This library is part of the AuthLib, we are overwriting this class to add a new field.
+ */
+
+public record KeyPairResponse1_19_0() {}
```
Comment (FlorianMichael) [inline src/main/java/de/florianmichael/viafabricplus/definition/authlib/KeyPairResponse1_19_0.java:8]: Delete this class and revert the whole key signing fix like it was in VFP 1.20.1

File: src/main/java/de/florianmichael/viafabricplus/injection/access/IKeyPairResponse.java [hunk 3]
```diff
@@ -5,4 +5,6 @@
 public interface IKeyPairResponse {
 
     ByteBuffer viafabricplus_getLegacyPublicKeySignature();
+
+    void viafabricplus_setLegacyPublicKeySignature(final ByteBuffer signature);
```
Comment (FlorianMichael) [inline src/main/java/de/florianmichael/viafabricplus/injection/access/IKeyPairResponse.java:9]: Delete this and revert the whole key signing fix like it was in VFP 1.20.1

File: src/main/java/de/florianmichael/viafabricplus/injection/mixin/fixes/authlib/MixinYggdrasilUserApiService.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.net.URL;
+
+/*
+Workaround for https://github.com/ViaVersion/ViaFabricPlus/issues/258
+ */
+
+@Mixin(value = YggdrasilUserApiService.class, remap = false)
+public class MixinYggdrasilUserApiService {
```
Comment (FlorianMichael) [inline src/main/java/de/florianmichael/viafabricplus/injection/mixin/fixes/authlib/MixinYggdrasilUserApiService.java:39]: Delete this class and revert the whole key signing fix like it was in VFP 1.20.1

File: src/main/java/de/florianmichael/viafabricplus/mappings/PackFormatsMappings.java [hunk 5]
```diff
@@ -33,6 +33,7 @@ public class PackFormatsMappings {
     private final static Map<Integer, GameVersion> protocolMap = new HashMap<>();
 
     public static void load() {
+        registerVersion(VersionEnum.r1_20_2, 18, "1.20.2");
```
Comment (FlorianMichael) [inline src/main/java/de/florianmichael/viafabricplus/mappings/PackFormatsMappings.java:36]: The snapshot is missing here

File: README.md [hunk 6]
```diff
@@ -129,7 +128,7 @@ dependencies {
     <dependency>
         <groupId>de.florianmichael</groupId>
         <artifactId>viafabricplus</artifactId>
-        <version>2.8.2</version> <!-- Get latest version from releases -->
+        <version>2.8.7</version> <!-- Get the latest version from releases -->
```
[the remark this answers is not in the document]
Comment (HowardZHY) [reply README.md:131]: The latest was tip by IDEA: A determiner may be missing

[the remark this answers is not in the document]
Comment (FlorianMichael) [reply README.md:131]: Fair enough, I will sync this change to the main branch as well.

File: gradle.properties [hunk 7]
```diff
@@ -9,31 +9,31 @@ loader_version=0.14.22
...
 
 # lenni0451 libs
-reflect_version=1.2.2
+reflect_version=1.2.4
 
 # other libs
 mod_menu_version=7.2.1
 netty_codec_http_version=4.1.96.Final
-mixin_extras_version=0.2.0-beta.9
+mixin_extras_version=0.2.0-rc.5
```
Comment (FlorianMichael) [inline gradle.properties:39]: There is 0.2.0 as release.

File: src/main/java/de/florianmichael/viafabricplus/injection/mixin/jsonwebtoken/MixinDefaultCompressionCodecResolver.java [hunk 8]
```diff
@@ -1,41 +0,0 @@
-/*
```
[the remark this answers is not in the document]
Comment (HowardZHY) [reply src/main/java/de/florianmichael/viafabricplus/injection/mixin/jsonwebtoken/MixinDefaultCompressionCodecResolver.java:1]: Dep version using by Latest ViaBedrock no longer has it

File: src/main/java/de/florianmichael/viafabricplus/injection/mixin/compat/ipnext/MixinAutoRefillHandler_ItemSlotMonitor.java [hunk 9]
```diff
@@ -32,21 +32,22 @@
...
+@SuppressWarnings("all")
 @Pseudo
 @Mixin(targets = "org.anti_ad.mc.ipnext.event.AutoRefillHandler$ItemSlotMonitor")
 public class MixinAutoRefillHandler_ItemSlotMonitor {
 
     @Shadow
     public int currentSlotId;
 
-    @Inject(method = { "checkHandle", "checkShouldHandle" }, at = @At("HEAD"), cancellable = true)
+    @Inject(method = { "checkHandle", "checkShouldHandle" }, at = @At("HEAD"), cancellable = true, remap = false)
```
Comment (FlorianMichael) [inline src/main/java/de/florianmichael/viafabricplus/injection/mixin/compat/ipnext/MixinAutoRefillHandler_ItemSlotMonitor.java:43]: Just add the remap = false to the class header (Line 37) and remove it from all methods

**Pull request merged**